### ISSUE TITLE
Host Lyricify as a local webpage or webapp so it can be integrated into Home Assistant and any device with a screen

### ISSUE THREAD

**Issue by AnshulJ999** [WXRIW/Lyricify-App#412]
Hi, I am looking for a way to host Lyricify as a local webpage (say, localhost:1234), so it can be viewed on any device with a screen in the home network. For example I could open a browser on my tablet to this page and view the displayed lyrics. 

My main purpose is to add an iFrame card to Home Assistant and be able to view real-time synced lyrics there. I am working on a lyrics app to do this myself: https://github.com/AnshulJ999/SyncLyrics

But Lyricify is already very mature and well-developed. If it could be hosted, this would be really useful. 

Thanks.

**Comment by WXRIW** [comment]
I am not familiar with web development, nor do I currently have any plans for it. Lyricify 4 is a traditional Win32 application, so it is difficult to port to the web, making your suggestion hard to implement for Lyricify.

As for cross-platform support, you can always use other products in the Lyricify family. With Lyricify's full platform coverage and the powerful synchronization capability of Spotify Connect, you can view lyrics in sync across all your devices.

**Comment by WXRIW** [comment]
Since there's no further replies, this issue is considered to be closed now. If you have any other questions or suggestions, feel free to reopen this issue.

**Comment by AnshulJ999** [comment]
> Since there's no further replies, this issue is considered to be closed now. If you have any other questions or suggestions, feel free to reopen this issue.

Hi, thank you for the quick reply. I understand. I've been testing both Lyricify 4 and Mobile, and I must say this is absolutely amazing work. Especially Lyricify 4; the best lyrics app I could not even think of. 

I have a very niche use case (Home Assistant Dashboard) that requires my webpage solution, so I have been working a fork an app: https://github.com/AnshulJ999/SyncLyrics

I have gotten it to a useable condition; it works for me at least. 

However, I would very much appreciate if you could help me with some pointers:

1) Lyricify 4 has a lot of providers it seems, and is able to fetch lyrics for almost any song, even very niche ones that I listen to. 
2) My app only uses LRCLib for now. I tried to integrate NetEase and Spotify into it, but failed. 
3) Hence, if you could point me in the right direction for implementing more lyrics providers like NetEase, QQ, Musixmatch, etc., that would deeply appreciated. 

Basically once I can set up multiple lyrics providers in my app, its core would be complete. 

I understand your time is valuable, so any pointers would be helpful. Perhaps if you can share in brief how Lyricify is able to integrate these providers. 

Thank you!

**Comment by WXRIW** [comment]
Actually, the lyrics processing part is open sourced on GitHub. You can find them in WXRIW/Lyricify-Lyrics-Helper. The readme document are in Chinese, so use a translator if you need.
